File: SocketPython/code.py
import datetime

def is_num_type(number):
	if type(number) is type(0) or type(number) is type(0.1):
		return True
	return False

def stringChecksum(string):
	if type(string) is type(""):
		sum = 0

		for char in string:
			sum += ord(char)

		return sum
	return 0

def padding(token, string, bytes_):
	padding = ""
	if type(token) is type("") and type(bytes_) is type(0) and type(string) is type(""):
		places = bytes_ - len(string)
		if(places >= 0):
			for i in range(places):
				padding += token
			return padding + string
		elif places < 0:
			return False
	elif type(token) is type("") and type(bytes_) is type(0) and is_num_type(string):
		string = str(string)
		zeros = bytes_ - len(string)
		token = "0"
		if(zeros >= 0):
			for i in range(zeros):
				padding += token
			return padding + string
		elif places < 0:
			return False
	else:
		return False

def codeUpdateString(sensor, data):
	if type(sensor) is type("") and is_num_type(data):
		code = "u"
		date = datetime.datetime.now()
		try:
			code += padding(" ", sensor.lower(), 8)
			code += padding("0", data, 8)
			code += date.strftime("%H%M%S%d%m%Y")
			code += str(stringChecksum(code))
		except TypeError:
			return False
		return code
	return False

def codeObserverString(observer, sensor):
	if type(observer) is type("") and type(sensor) is type(""):
		code = "r"
		try:
			code += padding(" ", observer.lower(), 8)
			code += padding(" ", sensor.lower(), 8)
			code += str(stringChecksum(code))
		except TypeError:
			return False
		return code
	return False

def codeFromLog(log, b=[8,8,6,8]):
	if type(log) is type(""):
		code = "u"
		parts = log.split(",")
		parts[-1] = parts[-1].replace("-", "")
		parts[-2] = parts[-2].replace(":", "")

		for i in range(len(parts)):
			print(parts[i])
			parts[i] = padding(" ", parts[i].lower().strip(), b[i])
			print(parts[i])

		for part in parts:
			code += part

		return code

File: SocketRuby/code.rb

require 'date'

def stringChecksum(string)
	if string.is_a? String
		sum = 0

		string.split("").each do |char|
			sum += char.ord
		end
		return sum
	end
	return 0
end

def padding(token, string, bytes)
	padding = ""
	if token.is_a? String and string.is_a? String and bytes.is_a? Integer
		places = bytes - string.length
		if places > 0
			for i in 0..places-1
				padding += token
			end
			return padding + string
		elsif places == 0
			return string
		elsif places < 0
			return false
		end
	elsif string.is_a? Numeric and bytes.is_a? Numeric
		string = string.to_s
		zeros = bytes - string.length
		token = "0"
		if zeros > 0
			for i in 0..zeros-1
				padding += token
			end
			return padding + string
		elsif zeros == 0
			return string
		elsif places < 0
			return false
		end
	else
		return false
	end
end

def codeUpdateString(sensor, data)
	if sensor.is_a? String and data.is_a? Numeric
		code = "u"
		date = DateTime.now

		code += padding(" ", sensor.downcase, 8)
		code += padding("0", data, 8)
		code += date.strftime("%H%M%S%d%m%Y")
		code += padding("0",stringChecksum(code), 4)

		return code
	end
	return false
end

def codeObserverString(observer, sensor)
	if observer.is_a? String and sensor.is_a? String
		code = "r"
		code += padding(" ", observer.downcase, 8)
		code += padding(" ", sensor.downcase, 8)
		code += stringChecksum(code).to_s
		return code
	end
	return False
end

def codeFromLog(log, b=[8,8,6,8])
	if log.is_a? String
		code = "u"
		parts = log.split(",")
		parts[-1].tr!("-", "")
		parts[-2].tr!(":", "")

		for i in 0..parts.length-1
			parts[i] = padding(" ", parts[i].downcase.strip, b[i])
		end
		for parts in parts
			code += parts
		end
		code += padding("0", stringChecksum(code), 4).to_s

		return code
	end
	return false
end
